fog/view/load-test/src/main.rs

```diff
@@ -41,79 +35,173 @@ struct Config {
 
     /// Grpc retry config
     #[clap(flatten)]
     pub grpc_retry_config: GrpcRetryConfig,
+
+    /// Max number of requests before the test ends (without ctrl-c)
+    #[clap(long, default_value = "0", env = "MC_MAX_REQUESTS")]
+    pub max_requests: u64,
+}
+
+/// Metrics that we aggregate for the load test
+#[derive(Clone, Debug, Default)]
+pub struct Counters {
+    /// Number of requests made to fog view
+    pub num_requests: u64,
+    /// Number of errors resulting from those requests
+    pub num_errors: u64,
+    /// Total latency of these requests, in milliseconds
+    pub total_millis_latency: u64,
+}
+
+impl Counters {
+    fn avg_latency(&self) -> f64 {
+        if self.num_requests == 0 {
+            0f64
+        } else {
+            (self.total_millis_latency / self.num_requests) as f64 / 1000f64
+        }
+    }
+}
+
+impl Display for Counters {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        write!(
+            f,
+            "{{ {} requests, {} errors, {} avg latency (seconds) }}",
+            self.num_requests,
+            self.num_errors,
+            self.avg_latency()
+        )
+    }
+}
+
+impl core::ops::Sub for Counters {
+    type Output = Self;
+    fn sub(self, other: Self) -> Self {
+        Self {
+            num_requests: self.num_requests - other.num_requests,
+            num_errors: self.num_errors - other.num_errors,
+            total_millis_latency: self.total_millis_latency - other.total_millis_latency,
+        }
+    }
 }
 
 fn worker_thread(
     uri: String,
     grpc_retry_config: GrpcRetryConfig,
-    account_key: AccountKey,
     num_search_keys: usize,
-    num_reqs: Arc<AtomicU64>,
+    counters: Arc<Mutex<Counters>>,
+    stop_requested: Arc<AtomicBool>,
     logger: Logger,
 ) {
     let mut fog_view_client = build_fog_view_conn(&uri, grpc_retry_config, &logger);
 
     let resp = fog_view_client
         .request(0, 0, Default::default())
+        .map_err(|err| {
+            stop_requested.store(true, Ordering::SeqCst);
+            err
+        })
         .expect("request");
     let rng_record = &(resp.rng_records[0]);
-    let rng = VersionedKexRng::try_from_kex_pubkey(
-        &rng_record.pubkey,
-        &account_key.default_subaddress_view_private(),
-    )
-    .expect("kex");
+    let private_key = FromRandom::from_random(&mut rand::thread_rng());
+    let rng = VersionedKexRng::try_from_kex_pubkey(&rng_record.pubkey, &private_key).expect("kex");
     let search_keys = rng.take(num_search_keys).collect::<Vec<Vec<u8>>>();
 
-    loop {
-        let _resp = fog_view_client
-            .request(0, 0, search_keys.clone())
-            .expect("request");
-        num_reqs.fetch_add(1, Ordering::SeqCst);
+    while !stop_requested.load(Ordering::SeqCst) {
+        let start = Instant::now();
+        let result = fog_view_client.request(0, 0, search_keys.clone());
+        let duration = Instant::now().duration_since(start);
```

Comment:
I think this should work and is shorter
```suggestion
        let duration = start.elapsed();
```